test/dotnet-build.Tests/BuildOutputTests.cs

```diff
@@ -0,0 +1,160 @@
+    {
+        private readonly string _testProjectsRoot;
+        private readonly string[] _runtimeFiles =
+        {
+            "TestApp" + FileNameSuffixes.DotNet.DynamicLib,
+            "TestApp" + FileNameSuffixes.DotNet.ProgramDatabase,
+            "TestApp" + FileNameSuffixes.CurrentPlatform.Exe,
+            "TestApp" + FileNameSuffixes.Deps,
+            "TestLibrary" + FileNameSuffixes.DotNet.DynamicLib,
+            "TestLibrary" + FileNameSuffixes.DotNet.ProgramDatabase
+        };
+        private readonly string[] _appCompileFiles =
+        {
+            "TestApp" + FileNameSuffixes.DotNet.DynamicLib,
+            "TestApp" + FileNameSuffixes.DotNet.ProgramDatabase
+        };
+        private readonly string[] _libCompileFiles =
+        {
+            "TestLibrary" + FileNameSuffixes.DotNet.DynamicLib,
+            "TestLibrary" + FileNameSuffixes.DotNet.ProgramDatabase,
+        };
+
+        public BuildOutputTests()
+        {
+            _testProjectsRoot = Path.Combine(AppContext.BaseDirectory, @"TestProjects");
+        }
+
+        private void PrepareProject(out TempDirectory root, out TempDirectory testAppDir, out TempDirectory testLibDir, out string runtime)
+        {
+            root = Temp.CreateDirectory();
+            var src = root.CreateDirectory("src");
+
+            testAppDir = src.CreateDirectory("TestApp");
+            testLibDir = src.CreateDirectory("TestLibrary");
+
+            // copy projects to the temp dir
+            CopyProjectToTempDir(Path.Combine(_testProjectsRoot, "TestApp"), testAppDir);
+            CopyProjectToTempDir(Path.Combine(_testProjectsRoot, "TestLibrary"), testLibDir);
+
+            var contexts = ProjectContext.CreateContextForEachFramework(
+                testLibDir.Path,
+                null,
+                PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers());
+            runtime = contexts.FirstOrDefault(c => !string.IsNullOrEmpty(c.RuntimeIdentifier))?.RuntimeIdentifier;
+        }
+
+        private string FormatPath(string input, string framework, string runtime)
+        {
+            return input.Replace("{fw}", framework).Replace("{rid}", runtime);
+        }
+
+        [Theory]
+        // global.json exists
+        [InlineData(true, null, null, "src/TestLibrary/bin/debug/{fw}", "src/TestApp/bin/debug/{fw}", "src/TestApp/bin/debug/{fw}/{rid}")]
+        [InlineData(true, "out", null, "src/TestLibrary/bin/debug/{fw}", "src/TestApp/bin/debug/{fw}", "out")]
+        [InlineData(true, null, "build", "build/src/TestLibrary/bin/debug/{fw}", "build/src/TestApp/bin/debug/{fw}", "build/src/TestApp/bin/debug/{fw}/{rid}")]
+        [InlineData(true, "out", "build", "build/src/TestLibrary/bin/debug/{fw}", "build/src/TestApp/bin/debug/{fw}", "out")]
+        //no global.json
+        //[InlineData(false, null, null, "src/TestLibrary/bin/debug/{fw}", "src/TestApp/bin/debug/{fw}", "src/TestApp/bin/debug/{fw}/{rid}")]
```

Comment:
We need dotnet test support to run tests somwhere without global.json to enable them either way.